src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -225,33 +290,46 @@ static void add_topic_config_if_requested(
           override_name,
           overrides_val,
           include_synonyms,
+          documentation,
           std::forward<Func>(describe_f));
     }
 }
 
+template<typename T>
+static ss::sstring maybe_print_tristate(const tristate<T>& tri) {
+    // Tristate.has_value also checks for disabled state
+    return tri.has_value() ? ssx::sformat("{}", tri.value()) : "-1";
+}
+
 template<typename T>
 static void add_topic_config(
   describe_configs_result& result,
   std::string_view default_name,
   const std::optional<T>& default_value,
   std::string_view override_name,
   const tristate<T>& overrides,
-  bool include_synonyms) {
-    std::optional<ss::sstring> override_value;
-    if (overrides.is_disabled()) {
-        override_value = "-1";
-    } else if (overrides.has_value()) {
-        override_value = ssx::sformat("{}", overrides.value());
+  bool include_synonyms,
+  std::optional<ss::sstring> documentation) {
+    // Make a copy of the default value because tristate contructor will move it
+    std::optional<T> value_copy(default_value);
+    tristate<T> value_as_tristate(value_copy);
+
+    // Wrap overrides in an optional because add_topic_config expects
+    // optional<S> where S = tristate<T>
+    std::optional<tristate<T>> overrides_opt;
+    if (overrides.has_value()) {
+        overrides_opt = std::make_optional(overrides);
     }
 
     add_topic_config(
       result,
       default_name,
-      default_value ? ssx::sformat("{}", default_value.value()) : "-1",
+      value_as_tristate,
```

Comment:
you can just insert wrapped item here:
`std::optional<T>(default_value)`

Reply:
> it take a non const parameter by value, so passing a const ref to it will make a copy, then the copy will be moved
OH! I mis-ready the signature and thought it was not making a copy. I'll change.